app/src/main/java/unipd/se18/ocrcamera/SearchResultActivity.java

```diff
@@ -0,0 +1,407 @@
+import android.widget.ArrayAdapter;
+import android.widget.AutoCompleteTextView;
+import android.widget.Button;
+import android.widget.Filter;
+import android.widget.ListView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import unipd.se18.ocrcamera.inci.Ingredient;
+import unipd.se18.ocrcamera.inci.IngredientsExtractor;
+
+
+/**
+ * This activity allow the user to manually search for ingredients, providing smart suggestions
+ * while the user is typing.
+ * Currently it is launched with an ACTION_SEARCH intent from the SearchView contained in the
+ * menu, inside ResultActivity.java
+ * @author Luca Moroldo g3
+ */
+public class SearchResultsActivity extends AppCompatActivity {
+
+    private static final String TAG = "SearchResultsActivity";
+
+    /**
+     * List used to show the results of a search
+     */
+    private ListView mIngredientsListView;
+
+    /**
+     * TexvView used to show messages when the result of a search is empty
+     */
+    private TextView mMessageTextView;
+
+    /**
+     * An extension of EditText that show suggestions with a dropdown list
+     */
+    private AutoCompleteTextView mAutoCompleteTextView;
+
+    /**
+     * Button that on click run a search and update mIngredientsListView with results
+     */
+    private Button mSearchButton;
+
+    /**
+     * Adapter used to update the dropdown suggestions list
+     */
+    private ArrayAdapter<String> dropDownListAdapter;
+
+    /**
+     * Used to find ingredients from the INCI db with a query
+     */
+    private IngredientsExtractor ingredientsExtractor;
+
+    /**
+     * Size of the dropdown suggestions list
+     */
+    private static final int MAX_DROPDOWN_SUGGESTIONS = 40;
+
+    /**
+     * Minimum query size to show suggestions
+     */
+    private static final int SUGGESTION_THRESHOLD = 2;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_search_results);
+
+        //get layout components
+        mIngredientsListView = findViewById(R.id.ingredients_list);
+        mMessageTextView = findViewById(R.id.message_text_view);
+        mAutoCompleteTextView = findViewById(R.id.ingredients_auto_complete_text_view);
+        mSearchButton = findViewById(R.id.auto_complete_text_view_button);
+
+        //mMessageTextView is used to show messages
+        mIngredientsListView.setEmptyView(mMessageTextView);
+
+        //get an instance of the ingredients extractor
+        ingredientsExtractor = InciSingleton.getInstance(getApplicationContext()).getIngredientsExtractor();
+
+
+        //add on item list click listener to show a dialog with ingredient's data
+        mIngredientsListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
```

Comment:
Equal to above. 
I have to read everything to understand what it does, while with a comment I could know it without having to read everything